app/src/test/java/com/codesoom/assignment/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,242 @@
+                assertThat(returnUser.getName()).isEqualTo(user.getName());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("update()")
+    class Describe_Update{
+
+        @Nested
+        @DisplayName("{id}에 해당하는 자원이 있고 , 수정할 정보가 비어있지 않다면")
+        class Context_ExistedIdAndNotEmptyBody{
+
+            private Long id = 1L;
+            private User oldUser;
+            private String updateContent;
+            private User updateUser;
+
+            @BeforeEach
+            void setUp() throws JsonProcessingException {
+                oldUser = addNewUser(command , id);
+                updateContent = "{\"name\":\"new name\",\"email\":\"new email\",\"password\":\"new password\"}";
+                updateUser = mapper.readValue(updateContent , User.class);
+            }
+
+            @AfterEach
+            void tearDown() {
+                repositoryClear(query , command);
+            }
+
+            @Test
+            @DisplayName("수정 후 수정된 정보를 반환한다.")
+            void It_UpdateUser() throws Exception {
+                MvcResult mvcResult = mvc.perform(patch("/user/" + oldUser.getId())
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(updateContent))
+                        .andExpect(status().isOk())
+                        .andReturn();
+                User returnUser = mapper.readValue(mvcResult.getResponse().getContentAsString() , User.class);
+                assertThat(returnUser.getName()).isEqualTo(updateUser.getName());
+                assertThat(returnUser.getEmail()).isEqualTo(updateUser.getEmail());
+            }
+        }
+
+        @Nested
+        @DisplayName("{id}에 해당하는 사용자가 존재하지 않는다면")
+        class Context_NotExistedId{
+
+            private Long id = 1L;
+            private String updateContent;
+
+            @BeforeEach
+            void setUp() {
+                repositoryClear(query , command);
+                updateContent = "{\"name\":\"new name\",\"email\":\"new email\",\"password\":\"new password\"}";
+            }
+
+            @Test
+            @DisplayName("자원이 존재하지 않는다는 예외를 던진다.")
+            void It_ThrowException() throws Exception {
+                mvc.perform(patch("/user/" + id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(updateContent))
+                        .andExpect(status().isNotFound());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("delete()")
+    class Describe_Delete{
+
+        @Nested
+        @DisplayName("{id}에 해당하는 사용자가 존재한다면")
+        class Context_ExistedId{
+
+            private User user;
+
+            @BeforeEach
+            void setUp() {
+                user = addNewUser(command ,1L);
+            }
+
+            @AfterEach
+            void tearDown() {
+                repositoryClear(query , command);
+            }
+
+            @Test
+            @DisplayName("삭제한다.")
```

Comment:
이렇게도 할 수 있겠죠. 
```suggestion
            @DisplayName("삭제를 수행하고, 삭제했음을 의미하는 응답을 돌려준다.")
```

하지만 제가 제안한 것보다 더 나은 방법도 있을 거에요. `NoContent`의 의미를 어떻게 구체적으로 표현할지에 대해서도 고민해 보세요.

Reply:
응답 코드를 명시하고 싶지만 응답 코드가 바뀐다면 설명도 바뀌어야 할 것 같네요
테스트 코드의 설명도 JavaDoc 같이 수정에 유연하게 작성해야 할까 궁금하네요 
실제로 구현 로직이 바뀐다면 테스트 코드를 먼저 수정하게 될테니 JavaDoc보다 자세하게 작성해도 되지 않을까 싶기도 합니다 ㅎㅎ
일단 응답 코드 설명을 추가해보도록 하겠습니다